src/installer/image/installer.go

```diff
@@ -17,35 +19,45 @@ import (
 
 type Properties struct {
 	ImageUri     string
+	ImageDigest  string
 	DockerConfig dockerconfig.DockerConfig
+	PathResolver metadata.PathResolver
 }
 
-func NewImageInstaller(fs afero.Fs, props *Properties) *imageInstaller {
-	return &imageInstaller{
+func NewImageInstaller(fs afero.Fs, props *Properties) *ImageInstaller {
+	return &ImageInstaller{
 		fs:    fs,
 		props: props,
 	}
 }
 
-type imageInstaller struct {
+type ImageInstaller struct {
 	fs    afero.Fs
 	props *Properties
 }
 
-func (installer *imageInstaller) InstallAgent(targetDir string) error {
-	log.Info("installing agent", "target dir", targetDir)
-	if err := installer.installAgentFromImage(targetDir); err != nil {
+func (installer *ImageInstaller) ImageDigest() string {
+	return installer.props.ImageDigest
+}
+
+func (installer *ImageInstaller) InstallAgent(targetDir string) error {
+	log.Info("installing agent from image")
+	_ = installer.fs.MkdirAll(targetDir, 0755)
+	if err := installer.installAgentFromImage(); err != nil {
 		_ = installer.fs.RemoveAll(targetDir)
 		return fmt.Errorf("failed to install agent: %w", err)
 	}
-	return symlink.CreateSymlinkForCurrentVersionIfNotExists(installer.fs, targetDir)
+	sharedDir := installer.props.PathResolver.AgentSharedBinaryDirForDigest(installer.props.ImageDigest)
+	return symlink.CreateSymlinkForCurrentVersionIfNotExists(installer.fs, sharedDir)
 }
 
-func (installer *imageInstaller) UpdateProcessModuleConfig(targetDir string, processModuleConfig *dtypes.ProcessModuleConfig) error {
-	return processmoduleconfig.UpdateProcessModuleConfig(installer.fs, targetDir, processModuleConfig)
+func (installer *ImageInstaller) UpdateProcessModuleConfig(targetDir string, processModuleConfig *dtypes.ProcessModuleConfig) error {
+	sourceDir := installer.props.PathResolver.AgentSharedBinaryDirForDigest(installer.props.ImageDigest)
+	return processmoduleconfig.CreateAgentConfigDir(installer.fs, targetDir, sourceDir, processModuleConfig)
 }
 
-func (installer *imageInstaller) installAgentFromImage(targetDir string) error {
+func (installer *ImageInstaller) installAgentFromImage() error {
+	defer func() { _ = installer.fs.RemoveAll(CacheDir) }()
```

Comment:
`error` is ignored anyway, so why not simply?
```suggestion
	defer installer.fs.RemoveAll(CacheDir)
```

Reply:
Ah, probably linter is complaining...and it's right.
We should discuss how to not ignore errors in `defer`.